### PR TITLE
Add operations sig charter and sig.yaml

### PR DESCRIPTION
**Pull request by HumairAK**
As discussed during the first sig operations meetings, here is the charter that was hashed out, along with the `sigs.yaml`. I've left some values as "TODO" to be filled in based on discussion, we still need to answer these questions: 

1. Who are the sig chairs? 
2. What subprojects ought to be added from the onset?

### CODE REVIEW

**Comment by HumairAK** [comment]
Ignore the owners comment, I'll add it in a separate pr to `operate-first/common` afterwards. 
I also added it to the jbook, didn't put a lot of thought in the toc structure, I'll leave that to a separate pr.

File: sig-operations/charter.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+## Scope
+
+The goal of this SIG is to enable management and operations for hardware and platform tooling within the Operate First Community Cloud (op1st)
+
+### In scope
+
+* Define what is part and not part of the Operate First Cloud platform offering
+* Communicate a platform roadmap
+* Day to day operations of platform
+* Operating and ensuring stability of
```
Comment (MichaelClifford) [inline sig-operations/charter.md:12]:
```
* Operating and ensuring stability of
```
Of what? is this a typo or are the items below suppose to be a sublist?

Comment (HumairAK) [reply sig-operations/charter.md:12]: yes this is a typo, thanks!

Comment (HumairAK) [reply sig-operations/charter.md]: fixed

File: sig-operations/charter.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+### In scope
+
+* Define what is part and not part of the Operate First Cloud platform offering
+* Communicate a platform roadmap
+* Day to day operations of platform
+* Operating and ensuring stability of:
+    * Platform management tools
+    * Services provided to workloads
+    * Platform component upgrades
```
Comment (sallyom) [inline sig-operations/charter.md:15]: `platform component upgrades` == upgrading cluster versions? 
`services provided to workloads` - such as? i'm trying to wrap my head around it is all
`platform management tools` == GitOps & Automation tools?

**Comment by sallyom** [comment]
let's discuss initial subprojects for SIG-operations [here](https://github.com/operate-first/community/issues/121)

File: sig-operations/charter.md (hunk 2 above)
Comment (HumairAK) [reply sig-operations/charter.md:15]: This was added by @durandom  so he can clarify but the way I understand it: 

> platform component upgrades == upgrading cluster versions? 

Yeah I think for the most part it's going to be clusters, though we're opting for broader terminology so as not to be limiting to only ocp clusters in the future.

>  services provided to workloads - such as? i'm trying to wrap my head around it is all 

This is probably referring to services like ODH, that are not directly related to operating the cloud and instead offer user's services like JupyterHub, Superset, etc. -- These are managed by the operations team. Though I think it we ought to distinguish between odh-operations (hopefully done by the ODH team) and cluster management, but maybe we can do that via  sub projects?

>  platform management tools == GitOps & Automation tools?

Yep, services like ArgoCD, ACM, Prow, etc.

Comment (sallyom) [reply sig-operations/charter.md:15]: service = OpenDataHub? AIOps?

Comment (sallyom) [reply sig-operations/charter.md:15]: thanks for the info! i'm getting a clearer idea now.

Comment (HumairAK) [reply sig-operations/charter.md:15]: hrmm maybe "Services provided to workloads" is a bit confusing, I'm understanding this as: 

"Management of regular services not directly tied to cluster management, for example ODH, Strimzi, Grafana, Openshift Pipelines, etc."

Comment (HumairAK) [reply sig-operations/charter.md:15]: @durandom can clarify

File: sig-operations/charter.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+* Procuring hardware and infrastructure
+
+## Stakeholders
+
+* User workload operations
+* Procuring hardware and infrastructure
+
+### Subproject Creation
+
+Sig Chairs are responsible for creation and removal of subprojects.
```
Comment (quaid) [inline sig-operations/charter.md:37]: Does this refer to the physical labor in the git repo, or the conceptual creation and removal?

If it's the latter, it seems like this is something that should be done by SIG consensus (both parts) and this reads like the SIG Chair has sole discretion to create or destroy a subproject.

File: sigs.yaml [hunk 4]
```diff
@@ -0,0 +1,25 @@
+sigs:
+- dir: sig-operations
+  name: Operations
+  mission_statement: >
+    The goal of this SIG is to enable management and operations for hardware and platform tooling within the Operate
```
Comment (quaid) [inline sigs.yaml:5]: It seems to me that `enable` is doing a LOT of work as a verb here, and it's not a very clear one in this case. It feels like we're looking e.g.:

```suggestion
    The goal of this SIG is to define and perform management and operations for hardware and platform tooling within the Operate
```

**Review by quaid — APPROVED** [review]
Looks very good overall. After we do the `sig-community` kickoff and similar PR, I'll note any new ideas we come up with for the template.

File: sig-operations/charter.md (hunk 2 above)
Comment (sallyom) [reply sig-operations/charter.md:15]: @HumairAK right, that's what I'm wondering
"Services made available to workloads"? 
"Platform services that are made available to workloads"?

File: sigs.yaml [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+  charter_link: charter.md
+  label: operations
+  leadership:
+    chairs:
+    - github: tbd  # TODO: fill in
+      name: tbd  # TODO: fill in
+      company: Red Hat
+  meetings:
+  - description: Data Science SIG meeting
```
[the remark this answers is not in the document]
Comment (sallyom) [reply sigs.yaml:16]: I was wondering the same!

File: sig-operations/charter.md (hunk 3 above)
Comment (anishasthana) [reply sig-operations/charter.md:37]: ++ @quaid . Conceptual creation and removal should be done based on consensus.

File: sigs.yaml [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    day: Monday
+    time: "10:00"
+    tz: ET (East Time)
```
Comment (anishasthana) [inline sigs.yaml:19]: Nit: East time isn't a thing.

File: sig-operations/charter.md (hunk 2 above)
Comment (durandom) [reply sig-operations/charter.md:15]: Platform components == CNV
Services provided == Observatorium, Camel-K
Platform management tools == ACM

I would put ODH into the user workload bucket. But since we say on bullet point 1

> Define what is part and not part of the Operate First Cloud platform offering

we could shrink it to 

```
* Communicate a platform roadmap
* Day to day operations of the platform
* Ensuring the stability of the platform
* Upgrading the platform
```

File: sig-operations/charter.md [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+* Services provided to workloads
+* Platform component upgrades
+* Coordinate hardware and infrastructure provisioning
+* Resolving platform level incidents
+* Providing GitOps framework for managing platform
+* Providing Automation Runbooks for self servicing repetitive operations tasks
+* PR / Code reviews pertaining to cluster/platform management
+* Documentation of all above
+* Ensure recording decisions via the ADR process
+* Ensure inclusivity and transparency
```
[the remark this answers is not in the document]
Comment (durandom) [reply sig-operations/charter.md:23]: +1

I would just scope it to the specific topics covered by sig-operations

```
* Mentor new and existing contributors to areas that are in scope of this SIG
```

File: sig-operations/charter.md (hunk 3 above)
Comment (durandom) [reply sig-operations/charter.md:37]: @quaid since this probably applies to other sigs, can make your suggestion a default in the charter? And then just remove this section here?

File: sigs.yaml (hunk 4 above)
Comment (durandom) [reply sigs.yaml:5]: Our way of thinking was, that the SIG does not perform stuff, but subprojects do. Might be nit-picking.
How about

```suggestion
    The goal of this SIG is to define and coordinate management and operations for hardware and platform tooling within the Operate
```

File: sigs.yaml (hunk 5 above)
[the remark this answers is not in the document]
Comment (durandom) [reply sigs.yaml:16]: 🍝

File: sig-operations/charter.md (hunk 3 above)
Comment (HumairAK) [reply sig-operations/charter.md:37]: Well, I figured _someone_ has to be ultimately responsible for coordinating subprojects, bringing creation to a vote, etc. and I figured that responsibility would land on the chairs. But I'll update to sig concensus.

**Comment by durandom** [comment]
> Who are the sig chairs?

Any volunteers? Otherwise we'll discuss during the next meeting.

File: sig-operations/charter.md (hunk 3 above)
Comment (HumairAK) [reply sig-operations/charter.md]: Updated, let me know if this is something you'd rather have it as a default, I can remove it from here. (This is something I noticed being mentioned in some sig charters explicitly in the k8s docs).

File: sig-operations/charter.md (hunk 2 above)
Comment (HumairAK) [reply sig-operations/charter.md]: sounds good to me, done.

File: sig-operations/charter.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (HumairAK) [reply sig-operations/charter.md:22]: >  Accountable for making contributing to operations equitable to contributing to other parts of the project 

@quaid could you maybe reword this, I'm not sure I understand. Are you saying the sig should, "ensure open sourcing of operations in such a way that there is a low barrier to entry for new contributors (with regards to operations)." ?? Not sure what you mean by "rest of the project" what project? You mean other sigs in operate first?

File: sigs.yaml (hunk 4 above)
Comment (HumairAK) [reply sigs.yaml:5]: Updated

**Comment by durandom** [comment]
/approve

I think this is a good start. let's get it in and we can always come back and make small adjustments

File: sig-operations/charter.md [hunk 8]
```diff
@@ -0,0 +1,37 @@
+# SIG Operations
+
+## Scope
+
+The goal of this SIG is to define and coordinate management and operations for hardware and platform tooling within the
```
Comment (billburnseh) [inline sig-operations/charter.md:5]: Would like to see more specific here, whether this SIG is intended to cover all varients of community clouds or is specific to the initial one that is OpenStack on bare metal based.

**Comment by HumairAK** [comment]
Can we pick a different time? The current suggestion (10am EST) conflicts with an existing meeting with ACM folks, and from what I recall @hpdempsey has a conflict or cannot make this time slot. How about Tuesday 10 AM Est?

File: sig-operations/charter.md [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+
+### Out of scope
+
+* User workload operations
+* Procuring hardware and infrastructure
+
+## Stakeholders
+
+* User workload operations
+* Procuring hardware and infrastructure
```
Comment (tumido) [inline sig-operations/charter.md:32]: This looks like a copy paste error to me. These are not stakeholders. Also, please correct me if I'm wrong, I don't think SIG requires stakeholders, because SIG is the stakeholder.

File: sigs.yaml [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+    The goal of this SIG is to enable management and operations for hardware and platform tooling within the Operate
+    First Community Cloud (op1st)
+
+  charter_link: charter.md
+  label: operations
+  leadership:
+    chairs:
+    - github: tbd  # TODO: fill in
+      name: tbd  # TODO: fill in
+      company: Red Hat
```
Comment (tumido) [inline sigs.yaml:14]: I nominate @HumairAK and @larsks (since there should be 2 chairs). Would you guys accept this proposal?

**Review by tumido — CHANGES_REQUESTED** [review]

**Comment by tumido** [comment]
> Can we pick a different time? The current suggestion (10am EST) conflicts with an existing meeting with ACM folks, and from what I recall @hpdempsey has a conflict or cannot make this time slot. How about Tuesday 10 AM Est?

Sure Tue 10AM EST works for me,  what about we hold at least one meeting at that time and if somebody is discontent with it, we do a doodle to propose new time?

File: sig-operations/charter.md (hunk 9 above)
Comment (HumairAK) [reply sig-operations/charter.md:32]: I'm getting the impression you are right, it seems wg have stakeholders which are sigs, sigs themselves don't have stakeholders. Seems strange to me, I would think other sigs/groups/orgs may have a stake in the sig :man_shrugging:

File: sigs.yaml (hunk 10 above)
Comment (larsks) [reply sigs.yaml]: Sure :)

Comment (HumairAK) [reply sigs.yaml]: I added these as placeholers for now to allow for generator to work w/o errors. They are not necessarily final, we can discuss further here or in next meeting.

File: OWNERS_ALIASES [hunk 11]
```diff
@@ -0,0 +1,7 @@
+aliases:
+  sig-operations-leads:
+    - HumairAK
+    - larsk
```
Comment (larsks) [inline OWNERS_ALIASES:4]: "larsks"

File: sig-operations/OWNERS [hunk 12]
```diff
@@ -0,0 +1,8 @@
+# See the OWNERS docs at https://go.k8s.io/owners
+
+reviewers:
+  - sig-operations
```
[the remark this answers is not in the document]
Comment (HumairAK) [reply sig-operations/OWNERS:4]: Ran the tool, needed generator files, went ahead and added those along with the PR, plus the generated files. I just took them from OSG repo and made some minor adjustments to the slack and mentions of k8s.

File: sigs.yaml [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+    tooling within the Operate First Community Cloud (op1st)
+
+  charter_link: charter.md
+  label: operations
+  leadership:
+    chairs:
+    - github: HumairAK
+      name: Humair Khan
+      company: Red Hat
+    - github: larsk
```
Comment (larsks) [inline sigs.yaml:15]: "larsks"

File: sigs.yaml [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+      name: Lars Kellog-Stedman
```
Comment (larsks) [inline sigs.yaml:16]: "Kellogg-Stedman"

File: sig-list.md [hunk 15]
```diff
@@ -0,0 +1,50 @@
...
+
+Each group's material is in its subdirectory in this project.
+
+When the need arises, a [new SIG can be created](sig-wg-lifecycle.md)
+
+### Special Interest Groups
+
+| Name | Label | Chairs | Contact | Meetings |
+|------|-------|--------|---------|----------|
+|[Operations](sig-operations/README.md)|operations|* [Humair Khan](https://github.com/HumairAK), Red Hat<br>* [Lars Kellog-Stedman](https://github.com/larsk), Red Hat<br>||* SIG Operations meeting: [Tuesdays at 10:00 EST (weekly)](tbd)<br>
```
Comment (larsks) [inline sig-list.md:37]: "larsks"

File: sig-operations/README.md [hunk 16]
```diff
@@ -0,0 +1,43 @@
...
+## Meetings
+* SIG Operations meeting: [Tuesdays at 10:00 EST](tbd) (weekly). [Convert to your timezone](http://www.thetimezoneconverter.com/?t=10:00&tz=EST).
+
+## Leadership
+
+### Chairs
+The Chairs of the SIG run operations and processes governing the SIG.
+
+* Humair Khan (**[@HumairAK](https://github.com/HumairAK)**), Red Hat
+* Lars Kellog-Stedman (**[@larsk](https://github.com/larsk)**), Red Hat
```
Comment (larsks) [inline sig-operations/README.md:26]: "larsks"

File: sig-operations/charter.md (hunk 8 above)
Comment (durandom) [reply sig-operations/charter.md:5]: I think being specific would just limit us upfront. This sig is responsible for all the platform component, including OpenStack, if we have some in the future...

File: sig-operations/README.md (hunk 16 above)
Comment (larsks) [reply sig-operations/README.md:26]: Also "Kellogg-Stedman"

**Comment by HumairAK** [comment]
Can I get some `/ lgtm` 's if we're happy with this iteration.

**Comment by HumairAK** [comment]
I added a commit to add generator/meta files, but if you'd just like to see sig-operations content see this commit [here](https://github.com/operate-first/community/pull/120/commits/da030fb82c27e033f310dfd60507c77daffa4235)

**Comment by durandom** [comment]
Looks like my LGTM did the trick :)
If we want to make amendments, lets do them in a follow up PR
